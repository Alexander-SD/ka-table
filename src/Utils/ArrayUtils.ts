export const getCopyOfArrayAndAddItem = (item: any, array: any[] = []): any[] => {
  return array.concat([item]);
};

export const getCopyOfArrayAndReplaceItem = (item: any, rowKey: any, array: any[]): any[] => {
  const newArray = [...array];
  const rowKeyValue = item[rowKey];
  const index = newArray.findIndex((i) => i[rowKey] === rowKeyValue);
  newArray.splice(index, 1, item);
  return newArray;
};
