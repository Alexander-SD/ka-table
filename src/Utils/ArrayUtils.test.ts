import { getCopyOfArrayAndAddItem } from './ArrayUtils';

describe('ArrayUtils', () => {
  it('getCopyOfArrayAndAddItem add item to array', () => {
    const array = [1, 2];
    const result = getCopyOfArrayAndAddItem(3, array);

    expect(result).not.toEqual(array);
    expect(result).toStrictEqual([1,2,3]);
  });

  it('getCopyOfArrayAndAddItem add item to empty array', () => {
    const result = getCopyOfArrayAndAddItem(1, undefined);    
    expect(result).toStrictEqual([1]);
  });
});
