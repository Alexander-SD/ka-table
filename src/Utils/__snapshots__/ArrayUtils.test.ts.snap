// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArrayUtils getCopyOfArrayAndReplaceItem 1`] = `
Array [
  Object {
    "column1": 1,
    "column2": 31,
  },
  Object {
    "column1": 2,
    "column2": 32,
  },
  Object {
    "column1": 3,
    "column2": 333,
  },
  Object {
    "column1": 4,
    "column2": 34,
  },
]
`;
